### PR TITLE
added login signup authentication

### PR DESCRIPTION
**Pull request by MonikaJhingan**
Pls review on an urgent basis.

### CODE REVIEW

File: src/reducer/signup-reducer.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export const signupReducer = (state, { type, payload }) => {
+    switch (type) {
+      case "FNAME":
+        return { ...state, firstName: payload };
+      case "LNAME":
```
Comment (rahulyadav139) [inline src/reducer/signup-reducer.js:5]: Type name could be better. Write the understandable names for better understanding.

**Review by rahulyadav139 — APPROVED** [review]

File: src/components/Login/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        setAuth({
+          user: { ...createdUser },
+          token: encodedToken,
+          auth: true,
+        });
+
+        localStorage.setItem("token", encodedToken);
+        navigate("/");
+      } else {
+        console.log("login failed");
```
Comment (xii-vi) [inline src/components/Login/Login.jsx:36]: _try to show this error message to the user_

File: src/components/Login/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        </label>
+
+        <div className="password">
+          <label htmlFor="remember">
+            <input name="checkbox" id="remember" type="checkbox" />
+            Remember Me
+          </label>
+          <a href="#" className="text-decorations password-forgot">
+            forgot your password?
+          </a>
```
Comment (xii-vi) [inline src/components/Login/Login.jsx:89]: _you could use `<p>` tag instead of `<a>` since you're not using href anyways._

**Review by xii-vi — APPROVED** [review]
- Clean Code
- Left a few sanity comments, go through them once before you merge.

File: src/reducer/signup-reducer.js (hunk 1 above)
Comment (MonikaJhingan) [reply src/reducer/signup-reducer.js:5]: ok will do.

File: src/components/Login/Login.jsx (hunk 2 above)
Comment (MonikaJhingan) [reply src/components/Login/Login.jsx:36]: ok

File: src/components/Login/Login.jsx (hunk 3 above)
Comment (MonikaJhingan) [reply src/components/Login/Login.jsx:89]: ok will do.Thnx

**Pull request merged**